extractors/pornhub/pornhub.go

```diff
@@ -0,0 +1,73 @@
+package pornhub
+
+import (
+	"encoding/json"
+	"github.com/iawia002/annie/downloader"
+	"github.com/iawia002/annie/request"
+	"github.com/iawia002/annie/utils"
+)
+
+type pornhubData struct {
+	DefaultQuality bool `json:"defaultQuality"`
+	Format string `json:"format"`
+	Quality string `json:"quality"`
+	VideoUrl string `json:"videoUrl"`
+}
+
+// Extract is the main function for extracting data
+func Extract(url string) ([]downloader.Data, error) {
+	var err error
+	html, err := request.Get(url, url, nil)
+	if err != nil {
+		return downloader.EmptyList, err
+	}
+	var title string
+	desc := utils.MatchOneOf(html, `<span class="inlineFree">(.+?)</span>`)
+	if desc != nil {
+		title = desc[1]
+	} else {
+		title = "pornhub video"
+	}
+
+	realURLs := utils.MatchOneOf(html, `"mediaDefinitions":\[(.+?)\],"isVertical"`)
+
+	var pornhubs []pornhubData
+	err = json.Unmarshal([]byte(`[` + realURLs[1] + `]`), &pornhubs)
+	if err != nil {
+		return downloader.EmptyList, err
+	}
+
+	//TODO add support for different quality
+	var realURL string
+	for _, downloadlink := range(pornhubs) {
+		if downloadlink.VideoUrl != "" {
+			realURL = downloadlink.VideoUrl
+			break
```

Comment:
So the first video in the list is the highest quality or?

Reply:
Yes, i think it's sorted like 1080p, 720p and goes on.